manim/mobject/value_tracker.py

```diff
@@ -57,6 +57,10 @@ def set_value(self, value):
     def increment_value(self, d_value):
         self.set_value(self.get_value() + d_value)
 
+    def __iadd__(self, d_value):
+        self.increment_value(d_value)
+        return self
```

Comment:
**Question : ** Why does it return self ? (I don't know this `__iadd__` stuff, so sorry if it's obvious)

Reply:
From the docs:

> These methods are called to implement the augmented arithmetic assignments (+=, -=, *=, @=, /=, //=, %=, **=, <<=, >>=, &=, ^=, |=). These methods should attempt to do the operation in-place (modifying self) and return the result (which could be, but does not have to be, self).

https://docs.python.org/3/reference/datamodel.html?highlight=iadd#object.__iadd__

[An example](https://stackoverflow.com/questions/1047021/overriding-in-python-iadd-method)